client/ayon_core/hosts/substancepainter/plugins/publish/validate_ouput_maps.py

```diff
@@ -108,3 +113,43 @@ def process(self, instance):
                 message=message,
                 title="Missing output maps"
             )
+
+
+    def get_invalid_channels(self, instance, config):
+        """Function to get invalid channel(s) from export channel
+        filtering
+
+        Args:
+            instance (pyblish.api.Instance): Instance
+            config (dict): export config
+
+        Raises:
+            PublishValidationError: raise Publish Validation
+                Error if any invalid channel(s) found
+
+        Returns:
+            list: invalid channel(s)
+        """
+        creator_attrs = instance.data["creator_attributes"]
+        export_channel = creator_attrs.get("exportChannel", [])
+        tmp_export_channel = copy.deepcopy(export_channel)
+        invalid_channel = []
+        if export_channel:
+            for export_preset in config.get("exportPresets", {}):
+                if not export_preset.get("maps", {}):
+                    raise PublishValidationError(
+                        "No Texture Map Exported with texture set:{}.".format(
+                            instance.name)
+                    )
+                map_names = [channel_map["fileName"] for channel_map
+                             in export_preset["maps"]]
+                for channel in tmp_export_channel:
+                    found = False
+                    for map_name in map_names:
+                        if channel in map_name:
+                            found = True
+                            break  # Exit the inner loop once a match is found
+                    if not found:
+                        invalid_channel.append(channel)
```

Comment:
A `for...else` should be able to do the same but nicer.
```suggestion
                    # Check if channel is found in at least one map
                    for map_name in map_names:
                        if channel in map_name:
                            break
                    else:
                        invalid_channel.append(channel)
```